python/modules/task.py

```diff
@@ -54,69 +54,45 @@ def parse_task_arguments(self, env_local):
             dict_setts[key] = value
         return dict_setts
 
-    def update_dict(self, dict_out, dict_in, list_by_value, list_by_name):
-        for k, v in list_by_value.items(): #update by value
-            dict_out[k] = v
-            logger.debug("Dict_out new key (updated by value): %s", k)
-        for k, v in list_by_name.items(): #update by name
-            logger.debug("Dict_in (by names) key, value: %s, %s", k, v)
-            try:
-                value = dict_in[v]
-            except:
-                logger.error("Dictionary update_by_name error. Given key (%s) doesn't exist in dictionary", v)
-            dict_out[k] = value
-            logger.debug("Dict_out new key, value (updated by name): %s, %s", k, v)
-        logger.debug("Dict_out before concatenation: %s", dict_out.keys())
-        try:
-            dict_out = self.concatenation_name_nr(dict_out)
-        except:
-            logger.error("Dictionary concatenation error. Dictionary: %s concatenation failed.", dict_out.keys())
-        logger.debug("Dict_out after concatenation: %s", dict_out.keys())        
+    def update_env(self, dict_out, dict_in, variables):
+        for key, var in variables.items():
+            var_out = VAR.get_variable(dict_in)
+            dict_out[key] = var_out       
         return dict_out
 
-    def concatenation_name_nr(self, dict_out):      
-        #concatenation name.number
-        key_list = []
-        for k, v in dict_out.items():
-            if "." in k:
-                key = k.split(".")
-                key_list.append(key[0])
-        logger.debug("List of names to concatenate: %s", key_list)
-        if len(key_list) > 1:
-            temp_dict = dict_out.copy()
-            temp_dict2 = OrderedDict(sorted(dict_out.items()))
-            key_list = list(set(key_list))      
-            for i in range(len(key_list)):
-                value = []
-                for k, v in temp_dict2.items():
-                    if key_list[i] + "." in k:
-                        value.append(str(v))
-                        del temp_dict[k] # deleting name.number from temporary dictionary
-                value = "".join(value)
-                temp_dict[key_list[i]] = value
-            dict_out = temp_dict.copy()
+    def convert_to_var_dict(self, dict_in):

```

Comment:
Well it is as local as other task class functions (parse_task_arguments and update_env) except this function is called by many subtasks when update_env is called only by superclass TASK method. Would you like me to convert  all their names ($name$ -> _$name&)? I would rather change parse_task_arguments and update_env names and leave convert_to_var_dict as it is. However using "_" in python is just an convetion, won't prevent from calling function from other module anyway.